rlp/decode_test.go

```diff
@@ -437,6 +452,23 @@ var decodeTests = []decodeTest{
 		ptr:   new(recstruct),
 		error: "rlp: expected input string or byte for uint, decoding into (rlp.recstruct).Child.I",
 	},
+	{
+		input: "C0",
+		ptr:   new(invalidDotDot1),
+		error: "rlp: invalid struct tag \"..\" for rlp.invalidDotDot1.A (must be on last field)",
+	},
+	{
+		input: "C0",
+		ptr:   new(invalidDotDot2),
+		error: "rlp: invalid struct tag \"..\" for rlp.invalidDotDot2.B (field type is not slice)",
+	},
+
+	// struct tag ".."
+	{
+		input: "C3010203",
+		ptr:   new(dotdotRaw),
+		value: dotdotRaw{A: 1, DotDot: []RawValue{unhex("02"), unhex("03")}},
+	},
```

Comment:
Please add a test case where the tag is correctly set, but the input RLP contains something wrong. E.g. you want to parse into a list of strings and one of the elements it an int, not a string. Or something like this, to verify that mixing elements are caught and does not crash or similar.